### PR TITLE
fixes #897: Update versions of GHA's

### LINKED ISSUE #897: GitHub Action versions need to be updated for the repo
**Describe the bug**
The GitHub Actions being used need to be updated.

**To Reproduce**
1. Look at the most recent Monthly Release thread [run](https://github.com/BSData/wh40k-killteam/actions/runs/11116781105)
2. Scroll down to the 'Annotations'
4. See the warnings

For example, the `actions/checkout` is on version 2 and should be on version 4.

**Expected behavior**
No warnings should appear in the 'Annotations' section

**Device and data (please complete the following information):**
 - OS: Windows 10
 - Browser: Firefox

**Additional context**
I have a potential fix on the fork created. Specifically, on the `alerickson0:ae-update-ghas`. But, I need to test it first.

### PR DESCRIPTION
**Pull request by alerickson0**
Updating the version of GHA's being used. One reason is to address security concerns.

### CODE REVIEW

File: .github/workflows/ReleaseIssue.yml [hunk 1]
```diff
@@ -2,8 +2,9 @@
 name: Monthly Release Thread
 
 on:
-  schedule:
-   - cron: 0 0 1 * *  # Every 1st day of the month at 0000 – https://crontab.guru
+  # schedule:
+  #  - cron: 0 0 1 * *  # Every 1st day of the month at 0000 – https://crontab.guru
+  workflow_dispatch:
```
Comment (mal20k) [inline .github/workflows/ReleaseIssue.yml:7]: This is the monthly release, so we need it on a schedule of the first of the month, so removing the cron job is not what we want.

File: .github/workflows/ReleaseIssue.yml [hunk 2]
```diff
@@ -12,21 +13,28 @@ jobs:
...
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
 # Get current month for issue title
     - name: get-date
       id: date
-      run: echo "::set-output name=date::$(date +'%B %Y')"
+      run: |
+        release_thread_date="$(date +'%B %Y')"
+        echo "date=$release_thread_date" >> $GITHUB_OUTPUT
```
Comment (mal20k) [inline .github/workflows/ReleaseIssue.yml:23]: Nit: This can just be simplified as
```yaml
echo "date=$(date +'%B %Y')" >> $GITHUB_OUTPUT

**Review by mal20k — CHANGES_REQUESTED** [review]
Thanks for looking at this! A few things to change, but otherwise looks good!

File: .github/workflows/ReleaseIssue.yml (hunk 1 above)
Comment (alerickson0) [reply .github/workflows/ReleaseIssue.yml:7]: Ok, I'll set it back to scheduled. I was just trying to test out my changes first before submitting

**Pull request merged**